### PR TITLE
TransactionsControllerTest: catch uncaught errors

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

Slight update to test to ensure the error is thrown, if not, the test should fail 

## More information

Related to: https://github.com/MetaMask/metamask-extension/pull/14147

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

**Review by brad-decker — DISMISSED** [review]

File: app/scripts/controllers/transactions/index.test.js [hunk 1]
```diff
@@ -2200,14 +2200,17 @@ describe('Transaction Controller', function () {
...
-          'Cannot call updateTransactionGasFees on a transaction that is not in an unapproved state',
-        );
-      }
+      assert.throws(
+        () =>
+          txController.updateTransactionGasFees('4', {
+            maxFeePerGas: '0x0088',
+          }),
+        Error,
+        'Cannot call updateTransactionGasFees on a transaction that is not in an unapproved state',
```
Comment (adonesky1) [inline app/scripts/controllers/transactions/index.test.js:2209]: [nit] could we make this error `Cannot call updateTransactionGasFees on a transaction that is in an approved state` 
(not a double negative)

**Review by adonesky1 — DISMISSED** [review]
small nit but otherwise LGTM!

**Review by brad-decker — DISMISSED** [review]

File: app/scripts/controllers/transactions/index.test.js (hunk 1 above)
Comment (digiwand) [reply app/scripts/controllers/transactions/index.test.js]: @adonesky1 I agree these messages were confusing. Thanks for calling this out!

Currently, the transactions can have these statuses: 
``` 
  UNAPPROVED: 'unapproved',
  APPROVED: 'approved',
  REJECTED: 'rejected',
  SIGNED: 'signed',
  SUBMITTED: 'submitted',
  FAILED: 'failed',
  DROPPED: 'dropped',
  CONFIRMED: 'confirmed',
  PENDING: 'pending',
```

The calls should be blocked for other transactions that are not `approved`, like `dropped`. I updated the related function name and the error messages to hopefully add some clarity.

**Comment by adonesky1** [comment]
@digiwand looks like you need a rebase 😅

File: app/scripts/controllers/transactions/index.test.js [hunk 2]
```diff
@@ -2186,10 +2186,10 @@ describe('Transaction Controller', function () {
       assert.equal(result.userFeeLevel, 'high');
     });
 
-    it('throws error if status is not unapproved', function () {
+    it('should not update and should throw error if status is not unapproved', function () {
```
Comment (adonesky1) [inline app/scripts/controllers/transactions/index.test.js:2189]: again just a tiny nit for clarity (might just be for me haha)
```suggestion
    it('should not update and should throw error if status is not type "unapproved"', function () {
```

Comment (digiwand) [reply app/scripts/controllers/transactions/index.test.js]: Works for me! Thanks and updated

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by DELJUJAZI — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**